sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -185,10 +193,15 @@ def _transform(self, X, fitting):
         indices = np.frombuffer(indices, dtype=np.intc)
         indptr = np.frombuffer(indptr, dtype=np.intc)
         shape = (len(indptr) - 1, len(vocab))
-
         result_matrix = sp.csr_matrix((values, indices, indptr),
                                       shape=shape, dtype=dtype)
 
+        if self.drop_first_category:
+            vocab = {k: v for k, v in vocab.items()
+                     if k not in to_drop.values()}
+            feature_names = [feature for feature in feature_names
+                             if feature in vocab.keys()]
```

Comment:
you just want `in vocab` here, else you'll needlessly convert `vocab` to a list in each iteration.